FSW/src/util/memUtil.cpp

```diff
@@ -0,0 +1,250 @@
+const int WINDOW_LENGTH = 240;  // seconds, length of science data buffer 
+const int MAXFILES = 10;        // maximum number of files in flash storage
+
+const int ADC_CHIP_SELECT = 10; // chip select pin for ADC
+const int SPI_MAX_SPEED = 2000000; // clock speed of 2MHz for SPI
+
+// TODO: Update this with size of actual timestamp once it is known
+const int TIMESTAMP_SIZE = 1;   // array indices needed to store timestamp
+
+// contants for converting ADC bins to voltage
+const float ADC_BINS = 65536;       // bins, number of bins in ADC (2^16)
+const float ADC_MAX_VOLTAGE = 3.3;  // Volts, upper end of ADC voltage range
+const float ADC_MIN_VOLTAGE = 0.0;  // Volts, lower end of ADC voltage range
+
+// set number of measurements to store in science data buffer
+const int BUFFERSIZE = SAMPLING_RATE * WINDOW_LENGTH; // indices
+const int FILESIZE = BUFFERSIZE + TIMESTAMP_SIZE;
+
+// declare static variables so that they do not go away when we leave this module
+// whole purpose is to make this into a black-box module of sorts
+static int bufIdx = 0;               // index of next dataBuffer element to overwrite
+static float dataBuffer[BUFFERSIZE]; // create array to hold data buffer elements
+
+
+/* - - - - - - Module Driver Functions - - - - - - */
+
+/* - - - - - - dataProcessing - - - - - - *
+ * Usage:
+ *  Reads incoming data from ADC, measures against baseline,
+ *  sends to memory.
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  data from the ADC (voltage measurement)
+ */
+float dataProcessing()
+{
+    float voltage;
+
+    // begin an SPI connection
+    // TODO: we should probably begin the SPI connection in main to avoid duplicates
+    SPI.begin();  
+    pinMode(ADC_CHIP_SELECT, OUTPUT); // set ADC chip select pin to output
+
+    // access ADC Pin (SPI)
+    uint16_t photodiode16; 
+    SPI.beginTransaction(SPISettings(SPI_MAX_SPEED, MSBFIRST, SPI_MODE3)); //SPISettings(maxSpeed,dataOrder,dataMode)
+    digitalWrite(ADC_CHIP_SELECT, LOW);   // set Slave Select pin to low to select chip
+    photodiode16 = SPI.transfer16(0x0000);// transfer data, send 0 to slave, recieve data from ADC
+    digitalWrite(ADC_CHIP_SELECT, HIGH);  // set Slave Select pin to high to de-select chip
+    SPI.endTransaction();
+
+    // Convert ADC output to voltage by scaling over voltage range
+
+    //TODO: can we scale this by the actual digital voltage on the board?
+    voltage = photodiode16 / ADC_BINS * (ADC_MAX_VOLTAGE - ADC_MIN_VOLTAGE);
+
+    return voltage;
+}
+
+/* - - - - - - scienceMemoryHandling - - - - - - *
+ * Usage:
+ *  Takes data from processing and stores it in a buffer,
+ *  continuously overwriting old entries.
+ *  Copies buffer to long term memory when signaled by timing module.
+ * 
+ * Inputs:
+ *  none
+ *  
+ * Outputs:
+ *  None
+ */
+void scienceMemoryHandling()
+{    
+    if (dataProcessEvent.checkInvoked()){ // checks event status from timing module
+        float currentPhotodiodeVoltage = dataProcessing();
```

Comment:
Maybe rename this to just 'photodiodeVoltage'?
It's obvious what this variable is, but I think having 'current' and 'voltage' in the same variable name is a bit confusing

Reply:
Agreed